app/controllers/v1/base_controller.rb

```diff
@@ -22,18 +23,9 @@ def record_not_found(error)
     render json: {error: error.message}, status: :not_found
   end
 
-  def default_person_params
-    params.require(:person).permit(Person::ALL_AVAILABLE_ATTRIBUTES)
-  end
-
-  def person_from_params
-    default_person_params.map{|k,v| default_person_params[k] = v.try(:strip) || v }
-    Person.create_or_update(default_person_params) if default_person_params.present?
-  end
-
   def create_person_and_action(default_template_id: nil)
-    person_params = default_person_params
-    person = Person.create_or_update(person_params)
+    person_params = params.require(:person).permit(PersonConstructor.permitted_params)
+    person = PersonConstructor.new(person_params).build.tap(&:save) # temporary fix
```

Comment:
I'm not sure I follow this "temporary fix."   What issue were you running into?  Also, since this build method returns a person, can't you just call save without the `tap`?